addons/api/addon/serializers/host-catalog.js

```diff
@@ -38,19 +39,32 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { isPlugin, compositeType } = snapshot.record;
     const { options } = attribute;
-    // Nested conditionals for readability
-    // This deletes any fields that don't belong to the record type
+
+    // Delete any fields that don't belong to the compositeType
+    if (
+      isPlugin &&
+      json &&
+      options?.for &&
+      !options.for.includes(compositeType)
+    ) {
+      if (!fieldsByType[compositeType].includes(key)) {
```

Comment:
non-blocking: Could this if statement be included with the if statement above it?

Reply:
ah yes we can, updated!